### PR TITLE
Fix alert boxes using bulleted lists

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | Alert boxes were using bulleted lists for content, which produced a visual glitch in symfony pages on Firefox where half a bullet point was visible.<br>This PR replaces bulleted lists by paragraphs on symfony pages.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4327
| How to test?  | Look at alert boxes in symfony pages using Firefox

You can check out what alert boxes are supposed to look like here: https://cdn.rawgit.com/PrestaShop/prestashop-ui-kit/master/index.html#alerts

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8562)

### CODE REVIEW

**Comment by LittleBigDev** [comment]
Ok for the lists removal. But what if someone really needs to add a bullet list in his alert message ? What will be the styling ?

---

Reviewed 14 of 14 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6CYyqF6VvYvave73K:bl1dfzd)*

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig [hunk 1]
```diff
@@ -380,17 +380,17 @@
...
-    <li><span class="glyphicon glyphicon-exclamation-sign"></span> {{
-        error.messagePluralization is null
-        ? error.messageTemplate|trans(error.messageParameters, 'form_error')
-        : error.messageTemplate|transchoice(error.messagePluralization, error.messageParameters, 'form_error')
-        }}
-    </li>
-{%- endfor -%}
-    </ul>
-    {% if form.parent %}</span>{% else %}</div>{% endif %}
+            <p> {{
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig]: you are changing the semantic sense for "style" reasons. You should keep list but add a css class to remove the bullet => https://getbootstrap.com/docs/4.0/content/typography/#unstyled

File: src/PrestaShopBundle/Resources/views/Admin/layout.html.twig [hunk 2]
```diff
@@ -56,35 +56,35 @@ app.request.attributes.get('_route')
             <button type="button" class="close" data-dismiss="alert" aria-label="Close">
                 <span aria-hidden="true"><i class="material-icons">close</i></span>
             </button>
-            <ul class="alert-text">
+            <div class="alert-text">
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/layout.html.twig]: I disagree with this. More this make the list of notifications unreadable for blind people readers :/ please consider a solution where we keep list when it's a list

**Comment by eternoendless** [comment]
If a bulleted list is needed, it can be achieved like this:

```html
<div class="alert-text">
  <ul>
    <li>Item 1</li>
    <li>Item 2</li>
  </ul>
</div>
```

The main thing is that a block element (be it a `<p>` or a `<div>`) with class `alert-text` is needed, otherwise there's a problem with margins.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig, line 386 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6705FfL4-MhdIGFRR-r1-386:-L-6E3YDLsaX8-Lyq-GU:b-brbma0) ([raw file](https://github.com/prestashop/prestashop/blob/f0919cdb08e78b8db83cd3bb60aa6ce04daef56e/src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig#L386)):*
<details><summary><i>Previously, mickaelandrieu (Mickaël Andrieu) wrote…</i></summary><blockquote>

you are changing the semantic sense for "style" reasons. You should keep list but add a css class to remove the bullet => https://getbootstrap.com/docs/4.0/content/typography/#unstyled
</blockquote></details>

I thought about keeping the bulleted list here, but then I saw this:

```html
<ul class="list-unstyled">
```

So effectively, the `<li>`s were being used as `<p>`s, which would be semantically incorrect.

Anyway, I'm not sure if this block is used anywhere :thinking: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6DIXj7erhYr986usw:b-xn3zac)*

**Review by mickaelandrieu — DISMISSED** [review]
consider using the `list-unstyled` css class from Bootstrap 4 :)

**Comment by LittleBigDev** [comment]
IMO, hiding list's style is not semantically incorrect.
Semantic != styling

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/layout.html.twig, line 59 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6705FfL4-MhdIGFRS-r1-59:-L-6FOteG1TDuNT8Hl1b:bx5q1pv) ([raw file](https://github.com/prestashop/prestashop/blob/f0919cdb08e78b8db83cd3bb60aa6ce04daef56e/src/PrestaShopBundle/Resources/views/Admin/layout.html.twig#L59)):*
<details><summary><i>Previously, mickaelandrieu (Mickaël Andrieu) wrote…</i></summary><blockquote>

I disagree with this. More this make the list of notifications unreadable for blind people readers :/ please consider a solution where we keep list when it's a list
</blockquote></details>

IMO one or the other should not be mandatory. It should be possible to choose, depending on the context. A single alert in a paragraph, or multiple messages in a list.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6FECwrRK90xwDGG9o:b-8xmim8)*

**Comment by eternoendless** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/layout.html.twig, line 59 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6705FfL4-MhdIGFRS-r1-59:-L-6EoDSQUBVunqxcGR3:b4li47e) ([raw file](https://github.com/prestashop/prestashop/blob/f0919cdb08e78b8db83cd3bb60aa6ce04daef56e/src/PrestaShopBundle/Resources/views/Admin/layout.html.twig#L59)):*
<details><summary><i>Previously, mickaelandrieu (Mickaël Andrieu) wrote…</i></summary><blockquote>

I disagree with this. More this make the list of notifications unreadable for blind people readers :/ please consider a solution where we keep list when it's a list
</blockquote></details>

I don't think that a sequence of paragraphs that is **shown** as a sequence of paragraphs for sighted people should be read as a bulleted list for blind people. It wouldn't be the same semantically.

I believe it would be different if we had a title, like: "Some errors were found: <list here>". That would allow for a bulleted list for everyone.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562)*

**Comment by LittleBigDev** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/layout.html.twig, line 59 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6705FfL4-MhdIGFRS-r1-59:-L-6G-gDXc2QCv0wpnbg:b-k0uhh5) ([raw file](https://github.com/prestashop/prestashop/blob/f0919cdb08e78b8db83cd3bb60aa6ce04daef56e/src/PrestaShopBundle/Resources/views/Admin/layout.html.twig#L59)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

I don't think that a sequence of paragraphs that is **shown** as a sequence of paragraphs for sighted people should be read as a bulleted list for blind people. It wouldn't be the same semantically.

I believe it would be different if we had a title, like: "Some errors were found: <list here>". That would allow for a bulleted list for everyone.
</blockquote></details>

I agree. This is why choice should be possible. And bullet points should be styled (margins etc)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562)*

**Comment by mickaelandrieu** [comment]
> IMO one or the other should not be mandatory. It should be possible to choose, depending on the context. A single alert in a paragraph, or multiple messages in a list.

yes, but we can't use a list of paragraphs instead of list of list items, because semantically it wont be read the same way by browsers and specific readers. It's a detail I know, for me it doesnt block the merge if you guys know the impact of this change :)

**Comment by LittleBigDev** [comment]
List of paragraphs would be weird. But this should be possible, and styled :
```html
<ul class="alert-text">
    <li>Item 1</li>
    <li>Item 2</li>
</ul>
```

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6GvYHi84e6Sv0NnYi:bqgekx4)*

**Comment by eternoendless** [comment]
That won't work because of margins, you need to wrap it in a div

With div


Without


I'm not against having bulleted lists, I'm just saying that we should only used when it makes semantic sense.

There's no sense in having a bulleted list that is made out of a single element or that is shown as paragraphs for sighted people, _because that's not a list_.

What about showing it as a single paragraph when there's only one element and as a bulleted list (with visible bullet points) when there's more than one element?

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/layout.html.twig, line 59 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6705FfL4-MhdIGFRS-r1-59:-L-6GIQeJEjZBe3UlWms:b-nig8m6) ([raw file](https://github.com/prestashop/prestashop/blob/f0919cdb08e78b8db83cd3bb60aa6ce04daef56e/src/PrestaShopBundle/Resources/views/Admin/layout.html.twig#L59)):*
<details><summary><i>Previously, LittleBigDev (Bastien Bieri) wrote…</i></summary><blockquote>

I agree. This is why choice should be possible. And bullet points should be styled (margins etc)
</blockquote></details>

A choice _is_ possible, as I explained earlier, and it _will_ show bullet points if you wrap it in a div (otherwise the bullet points are half-hidden by margins).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6HXwnuHJpJ08diLOA:bgt4d9o)*

**Comment by LittleBigDev** [comment]
Can't the `<ul>` be the needed block element ? I don't get why we should add another block wrapper around it.
I agree with you for everything else :
```html
<div class="alert-text">
    Single alert message
</div>
<ul class="alert-text">
    <li>First alert message</li>
    <li>Second alert message</li>
</div>
```

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6JAb6Dzx2XoJuwQE0:b8n5768)*

**Comment by eternoendless** [comment]
* Now it's possible to have `<ul>`s directly instead of having to wrap them up
* We use `<ul>` if there's more than one element to show

It looks like this:


---

Review status: 12 of 19 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-6S-SozG7-SdSvuOiU:b-v9stii)*

**Comment by LittleBigDev** [comment]
Reviewed 7 of 7 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig, line 397 at r3](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6svE1uni-rYGaQDF2:-L-6svE1uni-rYGaQDF3:bfaadk) ([raw file](https://github.com/prestashop/prestashop/blob/dd260c52c7d6fff4a28ca8589b073c9c45f95722/src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig#L397)):*
> ```Twig
>         {%- else -%}
>             <div class="alert-text">
>             {%- for error in errors -%}
> ```

No need to loop through it, as there is only 1 element (or maybe the `for` is still more elegant ?)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562)*

**Comment by eternoendless** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig, line 397 at r3](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6svE1uni-rYGaQDF2:-L-7bl38Ni118tksgxbN:b9c38xy) ([raw file](https://github.com/prestashop/prestashop/blob/dd260c52c7d6fff4a28ca8589b073c9c45f95722/src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig#L397)):*
<details><summary><i>Previously, LittleBigDev (Bastien Bieri) wrote…</i></summary><blockquote>

No need to loop through it, as there is only 1 element (or maybe the `for` is still more elegant ?)
</blockquote></details>

The alternative would be something like

```twig
{%- set error = errors[0] -%}
<p> ... </p>
```

I don't know if that would be much better, really

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562)*

**Comment by LittleBigDev** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig, line 397 at r3](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-L-6svE1uni-rYGaQDF2:-L-AtYMC2n7_gU2byPyd:bo6apac) ([raw file](https://github.com/prestashop/prestashop/blob/dd260c52c7d6fff4a28ca8589b073c9c45f95722/src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_3_layout.html.twig#L397)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

The alternative would be something like

```twig
{%- set error = errors[0] -%}
<p> ... </p>
```

I don't know if that would be much better, really
</blockquote></details>

Indeed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562)*

**Comment by LittleBigDev** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-AthkZ54ljXSwneN_J:bnfp4nl)*

**Review by LittleBigDev — DISMISSED** [review]

**Comment by LittleBigDev** [comment]
:+1: 

---

Reviewed 1 of 7 files at r3, 5 of 5 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8562#-:-L-CqkGBXqCB2je8iS8z:b-8fx2u7)*

**Review by LittleBigDev — APPROVED** [review]

**Pull request merged**